intron_selection_code.sh

```diff
@@ -0,0 +1,47 @@
+# Set the GTF input file used for alignment of human samples
+gtf_input="Drosophila_melanogaster.BDGP6.90.gtf"
+
+
+# Script to extract and sort genomic features (exons, introns, and UTRs) from a GTF file.
+# Usage: ./bedtools.sh
+
+
+# Define output paths for BED files by replacing the extension in the input filename
+bed_gene="${gtf_input/.gtf/.gene.bed}"
+bed_exon="${gtf_input/.gtf/.exon.bed}"
+bed_5prime_utr="${gtf_input/.gtf/.5prime_utr.bed}"
+bed_3prime_utr="${gtf_input/.gtf/.3prime_utr.bed}"
+bed_intron="${gtf_input/.gtf/.intron.bed}"
+
+# Function to process genomic features and output to BED files
+process_features() {
+    local feature_type=$1
+    local feature_label=$2
+    local output_file=$3
+
+    echo "Processing $feature_type..."
+    cat "$gtf_input" | awk -v label="$feature_label" 'BEGIN {OFS="\t"} $3 == label {print $1, $4, $5, label, $10, $7}' | bedtools sort > "$output_file"
```

Comment:
There seems to be an issue with combining GTF and BED file formats for genomic intervals: The GTF is 1-based with right-end included, while BED is 0-based with right-end excluded. See e.g. https://tag.readthedocs.io/en/stable/formats.html or https://tidyomics.com/blog/2018/12/09/2018-12-09-the-devil-0-and-1-coordinate-system-in-genomics/.